syncano/connection.py

```diff
@@ -181,10 +182,26 @@ def make_request(self, method_name, path, **kwargs):
         # Encode request payload
         if 'data' in params and not isinstance(params['data'], six.string_types):
             params['data'] = json.dumps(params['data'])
-
         url = self.build_url(path)
         response = method(url, **params)
+        content = self.get_response_content(url, response)
+
+        if files:
+            params.pop('data')
+            params['headers'].pop('content-type')
```

Comment:
Shouldn't you have a content-type set though? Like` Content-Type: multipart/form-data`

Anyway, we should have some comments in code about that, because it's not obvious.